src/turn-command-manager.ts

```diff
@@ -0,0 +1,477 @@
+  }
+
+  private handleBallCommand(turnCommand: TurnCommand): boolean {
+    const { cursor, targets } = turnCommand;
+
+    if (isNullOrUndefined(cursor) || isNullOrUndefined(targets)) {
+      console.error("Error encountered when trying to throw Pokeball!");
+      console.error(turnCommand);
+      return false;
+    }
+
+    globalScene.unshiftPhase(new AttemptCapturePhase(targets[0] % 2, cursor));
+    return true;
+  }
+
+  private handlePokemonCommand(turnCommand: TurnCommand): boolean {
+    const { pokemon, cursor } = turnCommand;
+    if (isNullOrUndefined(cursor)) {
+      console.error("Error encountered when trying to switch Pokemon!");
+      console.error(turnCommand);
+      return false;
+    }
+
+    const switchType = turnCommand.args?.[0] ? SwitchType.BATON_PASS : SwitchType.SWITCH;
+    globalScene.unshiftPhase(
+      new SwitchSummonPhase(switchType, pokemon.getFieldIndex(), cursor, true, pokemon.isPlayer()),
+    );
+    return true;
+  }
+
+  private handleRunCommand(turnCommand: TurnCommand): boolean {
+    let runningPokemon = turnCommand.pokemon;
+    if (globalScene.currentBattle.double) {
+      const playerActivePokemon = globalScene.getField(true).filter((pokemon) => pokemon.isPlayer());
+
+      if (playerActivePokemon.length > 1) {
+        const fasterPokemon = playerActivePokemon.sort((a, b) => b.getStat(Stat.SPD) - a.getStat(Stat.SPD))[0];
+
+        const hasRunAway = playerActivePokemon.find((p) => p.hasAbility(Abilities.RUN_AWAY));
+        runningPokemon = hasRunAway ?? fasterPokemon;
+      }
+    }
+    globalScene.unshiftPhase(new AttemptRunPhase(runningPokemon.getFieldIndex()));
+    return true;
+  }
+
+  /**
+   * Shifts all {@linkcode Command.BALL | BALL}, {@linkcode Command.POKEMON | POKEMON},
+   * and {@linkcode Command.RUN | RUN} commands in the queue.
+   * Turn commands in the queue should be sorted with {@linkcode setTurnOrder}
+   * before this function is called.
+   */
+  private shiftNonFightCommands(): void {
+    while (this.turnCommands[0] && this.turnCommands[0].command !== BattleCommand.FIGHT) {
+      this.shiftNextCommand();
+    }
+  }
+
+  /**
+   * Applies the effects of {@linkcode BypassSpeedChanceAbAttr | Quick Draw}
+   * and {@linkcode BypassSpeedChanceModifier | Quick Claw} to all commands
+   * in the turn command queue. Quick Draw and Quick Claw each
+   * have an independent chance of allowing their respective Pokemon
+   * to bypass Speed, but if Quick Draw successfully applies, Quick
+   * Claw cannot also apply.
+   */
+  private applyBypassSpeedEffects(): void {
+    this.turnCommands.forEach((tc) => {
+      const { pokemon, move } = tc;
+      // Only apply to commands to use damaging moves
+      if (!move || allMoves[move.moveId].category === MoveCategory.STATUS) {
+        return;
+      }
+
+      applyAbAttrs(AbAttrFlag.BYPASS_SPEED_CHANCE, pokemon, false);
+      globalScene.applyModifiers(BypassSpeedChanceModifier, pokemon.isPlayer(), pokemon);
```

Comment:
This check seems correct for quick draw since it only applies to damaging moves, but I don't think it is the same for quick claw. The only thing I could find in quick claw's previous implementation is that it checks that the turn command is a fight command, but it doesn't seem to exclude status moves to me
https://github.com/Despair-Games/poketernity/blob/117a3140618c0d31f2b3a10439cc6af22d4ca440/src/modifier/modifier.ts#L1707-L1708

So it should be something like this I think

```suggestion
      // Only apply to fight commands
      if (!move) {
        return;
      }

      if (allMoves[move.moveId].isDamageMove()) {
        applyAbAttrs(AbAttrFlag.BYPASS_SPEED_CHANCE, pokemon, false);
      }
      globalScene.applyModifiers(BypassSpeedChanceModifier, pokemon.isPlayer(), pokemon);
```